RecoTracker/TkTrackingRegions/plugins/CandidatePointSeededTrackingRegionsProducer.h

```diff
@@ -0,0 +1,451 @@
+    if (m_mode == VERTICES_FIXED)  m_zErrorVetex     = regPSet.getParameter<double>("zErrorVetex");
+    m_nSigmaZBeamSpot = -1.;
+    if (m_mode == BEAM_SPOT_SIGMA)
+    {
+      m_nSigmaZBeamSpot = regPSet.getParameter<double>("nSigmaZBeamSpot");
+      if (m_nSigmaZBeamSpot < 0.)
+        edm::LogError ("CandidatePointSeededTrackingRegionsProducer")<<"nSigmaZBeamSpot must be positive for BeamSpotSigma mode!";
+    }
+  }
+  
+  ~CandidatePointSeededTrackingRegionsProducer() override {}
+    
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+
+    desc.add<std::string>("mode", "BeamSpotFixed");
+    desc.add<std::string>("seedingMode", "Candidate");
+
+    desc.add<edm::InputTag>("input", edm::InputTag(""));
+    edm::ParameterSetDescription descPoints;
+    descPoints.add<std::vector<double>> ("eta", {} ); 
+    descPoints.add<std::vector<double>> ("phi", {} ); 
+    desc.add<edm::ParameterSetDescription>("points", descPoints);
+
+    desc.add<int>("maxNRegions", 10);
+    desc.add<edm::InputTag>("beamSpot", edm::InputTag("hltOnlineBeamSpot"));
+    desc.add<edm::InputTag>("vertexCollection", edm::InputTag("hltPixelVertices"));
+    desc.add<int>("maxNVertices", 1);
+
+    desc.add<double>("ptMin", 0.9);
+    desc.add<double>("originRadius", 0.2);
+    desc.add<double>("zErrorBeamSpot", 24.2);
+    desc.add<double>("deltaEta", 0.5);
+    desc.add<double>("deltaPhi", 0.5);
+    desc.add<double>("deltaEta_Cand", -1.);
+    desc.add<double>("deltaPhi_Cand", -1.);
+    desc.add<double>("deltaEta_Point", -1.);
+    desc.add<double>("deltaPhi_Point", -1.);
+    desc.add<bool>("precise", true);
+
+    desc.add<double>("nSigmaZVertex", 3.);
+    desc.add<double>("zErrorVetex", 0.2);
+    desc.add<double>("nSigmaZBeamSpot", 4.);
+
+    desc.add<std::string>("whereToUseMeasurementTracker", "ForSiStrips");
+    desc.add<edm::InputTag>("measurementTrackerName", edm::InputTag(""));
+
+    desc.add<bool>("searchOpt", false);
+
+    // Only for backwards-compatibility
+    edm::ParameterSetDescription descRegion;
+    descRegion.add<edm::ParameterSetDescription>("RegionPSet", desc);
+
+    descriptions.add("candidatePointSeededTrackingRegionsFromBeamSpot", descRegion);
+  }
+
+
+  std::vector<std::unique_ptr<TrackingRegion> > regions(const edm::Event& e, const edm::EventSetup& es) const override
+  {
+    std::vector<std::unique_ptr<TrackingRegion> > result;
+
+    // pick up the candidate objects of interest
+    edm::Handle< reco::CandidateView > objects;
+    size_t n_objects = 0;
+    size_t n_points = 0;
+
+    if(m_seedingMode == CANDIDATE_SEEDED || m_seedingMode == CANDIDATE_POINT_SEEDED){
+      e.getByToken( token_input, objects );
+      n_objects = objects->size();
+      if (n_objects == 0) return result;
+    }
+
+    if(m_seedingMode == POINT_SEEDED || m_seedingMode == CANDIDATE_POINT_SEEDED){
+      n_points = etaPoints.size();
+      if (n_points == 0) return result;
```

Comment:
I would move this check to the constructor and throw an exception in this case, because it makes no sense to have this code always produce 0 regions.